lib/force.go

```diff
@@ -1153,7 +1153,10 @@ func (f *Force) _coerceHttpError(res *http.Response, body []byte) error {
 	} else {
 		var errors ForceErrors
 		if err := internal.JsonUnmarshal(body, &errors); err != nil && !sessionExpired {
-			return err
+			var errors map[string]interface{}
```

Comment:
It doesn't look like anything happens with this error result?

Reply:
If we can unmarshal the body into a `map[string]interface{}`, we return the fallback error containing the body.

For example, if we get a 400 status code and the body is
```
{
  "status" : 400,
  "message" : "I do not understand",
  "extra" : { },
  "code" : "bad_input"
}
```

the error returned is
```
unhandled error: 400 {
  "status" : 400,
  "message" : "PartnerIdentifier is invalid",
  "extra" : { },
  "code" : "invalid_partner"
}
```

instead of
```
error unmarshaling json: json: cannot unmarshal object into Go value of type lib.ForceErrors. first 10 characters: {
  "statu
```

Reply:
Oh, I see. Maybe we should just `return fmt.Errorf("unhandled error: %d %s", res.StatusCode, string(body))` here instead? Doesn't seem like we want to error if the parsing fails, and also, the code path here is pretty confusing (throwing away the parse result and depending on behavior about 12 lines later to get a fallback error).

Reply:
> Oh, I see. Maybe we should just `return fmt.Errorf("unhandled error: %d %s", res.StatusCode, string(body))` here instead? Doesn't seem like we want to error if the parsing fails, and also, the code path here is pretty confusing (throwing away the parse result and depending on behavior about 12 lines later to get a fallback error).

Good idea.  Updated in 1e5bccea9dc98e91740d681a84319a1f83529002.